service/src/main/kotlin/io/provenance/digitalcurrency/consortium/frameworks/MigrationQueue.kt

```diff
@@ -0,0 +1,64 @@
+package io.provenance.digitalcurrency.consortium.frameworks
+
+import io.provenance.digitalcurrency.consortium.bankclient.BankClient
+import io.provenance.digitalcurrency.consortium.config.CoroutineProperties
+import io.provenance.digitalcurrency.consortium.config.logger
+import io.provenance.digitalcurrency.consortium.domain.MigrationRecord
+import io.provenance.digitalcurrency.consortium.domain.toAlertRequest
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.springframework.context.event.EventListener
+import org.springframework.stereotype.Component
+import java.util.UUID
+
+class MigrationDirective(
+    override val id: UUID
+) : Directive()
+
+class MigrationOutcome(
+    override val id: UUID
+) : Outcome()
+
+@Component
+class MigrationQueue(
+    coroutineProperties: CoroutineProperties,
+    private val bankClient: BankClient
+) : ActorModel<MigrationDirective, MigrationOutcome> {
+
+    private val log = logger()
+
+    @EventListener(DataSourceConnectedEvent::class)
+    fun startProcessing() {
+        log.info("start migration alerting framework")
+        start()
+    }
+
+    override val numWorkers: Int = coroutineProperties.numWorkers.toInt()
+    override val pollingDelayMillis: Long = coroutineProperties.pollingDelayMs.toLong()
+
+    override suspend fun loadMessages(): List<MigrationDirective> =
+        transaction {
+            MigrationRecord.findPending().map { MigrationDirective(it.id.value) }
+        }
+
+    override fun processMessage(message: MigrationDirective): MigrationOutcome {
+        transaction {
+            val migration = MigrationRecord.findForUpdate(message.id).first()
+            if (migration.sent != null)
+                return@transaction // already processed
+
+            bankClient.persistAlert(migration.toAlertRequest())
```

Comment:
There's not really a need for this transaction to span the api call. It would have the same output with two separate transactions.

Reply:
good catch, thanks.